sqlair.go

```diff
@@ -561,7 +542,7 @@ func (tx *TX) Query(ctx context.Context, s *Statement, inputArgs ...any) *Query
 		return &Query{ctx: ctx, err: ErrTXDone}
 	}
 
-	sqlstmt, err := tx.db.prepareStmt(ctx, tx.sqlconn, s)
+	sqlstmt, err := tx.db.prepareStmt(ctx, s)
```

Comment:
What about this?
- `prepareStmt` checks `dbStats` to see if `MaxOpenConnections` is set to 1.
- If it is, we return a typed error.
- If we catch that error here, we log a warning and just prepare against the transaction instead of the `db`, skipping the cache.

Reply:
That could work but only for that specific case. If we had `MaxOpenConnections` set to 2 and we created two transactions we are going to have the same problem. And the same applies to `n` connections and `n` transactions. My other concern is that `MaxOpenConnections` can be changed concurrently from another thread accessing the underlying sqldb which is going to lead to some bugs that are difficult to trace.

Another solution would be to check whether there is a max number of connections and issue a warning or even return an error. We could even require an environment variable to be set just to be explicit about the number of connections being a problem. The issue, once again, is the concurrency aspect.